docs/guide.md

```diff
@@ -0,0 +1,575 @@
+# Firebase Security and Rules Using the Bolt Compiler
+
+Firebase is secured using a JSON-formatted [Security and
+Rules](https://www.firebase.com/docs/security/guide/understanding-security.html) language. It
+is a powerful feature of Firebase, but can be error prone to write by hand.
+
+The Bolt compiler helps developers express the schema and authorization rules for their
+database using a familiar JavaScript-like language. The complete [language
+reference](language.md) describes the syntax. This guide introduces the concepts and
+features of Bolt along with a cookbook of common recipies.
+
+# Getting Started
+
+The Firebase Bolt compiler is a command-line based tool, written in node.js. You can install it
+using the [Node Package Manager](https://nodejs.org/en/download/):
+
+    $ npm install --global firebase-bolt
+
+You can use the Bolt compiler to compile the examples in this guide, and inspect the output
+JSON.
+
+## Default Firebase Permissions
+
+By default, Firebase has a permissive security model - this makes it easy to test your code
+(but is unsafe for production apps since anyone can read and overwrite all your data). In Bolt,
+these default permissions can be written as:
+
+[all_access.bolt](../samples/all_access.bolt)
+```javascript
+path / {
+  read() = true;
+  write() = true;
+}
+```
+
+This _path_ statement, defines read and write permissions to every part of your database
+(all paths beneath the root (`/`) of the database).  The `read` and `write` methods are defined
+using the syntax `read() = <expression>;`.  When the expression evaluates to `true`, reading
+(or writing) is allow at the given path location (and all children of that path).
+
+Use the Bolt compiler to convert this to Firebase JSON-formatted rules:
+
+    $ firebase-bolt < all_access.bolt
+
+```JSON
+{
+  "rules": {
+    ".read": "true",
+    ".write": "true"
+  }
+}
+```
+
+## How to Use Bolt in Your Application
+
+Bolt is (not yet) integrated into the online [Firebase Security and Rules
+Dashboard](https://www.firebase.com/account/). There are two ways to use Bolt to define rules
+for your application:
+
+1. Use the firebase-bolt command line tool to generate a JSON file from your Bolt file, and
+   then copy and paste the result into the Dashboard _Security and Rules_ section.
+2. Use the [Firebase Command Line](https://www.firebase.com/docs/hosting/command-line-tool.html)
+   tool.  If you have _firebase-bolt_ installed on your computer, you can set the `rules` property
+   in your [firebase.json](https://www.firebase.com/docs/hosting/guide/full-config.html) file
+   to the name of your Bolt file.  When you use the `deploy` command, the command line will
```

Comment:
When you issue the `firebase deploy` command, it will read and compile your Bolt file and upload the compiled JSON to your Firebase application.